File: LittleProjects/Scala/ScalatraDoobieCrud/scalatradoobiecrud/src/main/scala/com/github/gr1f0n6x/scalatra/doobie/crud/service/EntityService.scala
package com.github.gr1f0n6x.scalatra.doobie.crud.service

import com.github.gr1f0n6x.scalatra.doobie.crud.model.Entity
import com.github.gr1f0n6x.scalatra.doobie.crud.repository.EntityRepository

class EntityService extends Service[Entity, Long] {
  val repository: EntityRepository = EntityRepository()

  override def findOne(id: Long): Either[Throwable, Entity] = repository.findOne(id).attempt.unsafeRunSync()

  override def findAll(): Either[Throwable, Seq[Entity]] = repository.findAll().attempt.unsafeRunSync()

  override def save(a: Entity): Either[Throwable, Entity] = repository.save(a).attempt.unsafeRunSync()

  override def update(a: Entity): Either[Throwable, Int] = repository.update(a).attempt.unsafeRunSync()

  override def deleteOne(id: Long): Either[Throwable, Int] = repository.deleteOne(id).attempt.unsafeRunSync()

  override def deleteAll(): Either[Throwable, Int] = repository.deleteAll().attempt.unsafeRunSync()

  def findFirst5(): Either[Throwable, Seq[Entity]] = repository.findFirst5().attempt.unsafeRunSync()
}

object EntityService {
  def apply(): EntityService = new EntityService()
}

File: LittleProjects/Java/GuiceHibernatePostgresCrud/src/main/java/guice/hibernate/postgres/crud/service/EntityService.java
package guice.hibernate.postgres.crud.service;

import guice.hibernate.postgres.crud.model.Entity;
import guice.hibernate.postgres.crud.repository.EntityRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import javax.persistence.EntityManager;
import javax.persistence.EntityManagerFactory;
import javax.persistence.EntityTransaction;
import javax.persistence.Persistence;
import java.util.Collections;
import java.util.Optional;

public class EntityService implements Service<Entity, Long> {
    private Logger logger = LogManager.getLogger("entity-service");
    private EntityManagerFactory emf = Persistence.createEntityManagerFactory("PostgreSQL");

    @Override
    public Optional<Entity> findOne(Long aLong) {
        EntityManager em = emf.createEntityManager();
        EntityTransaction tx = em.getTransaction();

        try {
            tx.begin();
            Optional<Entity> e = new EntityRepository(em).findOne(aLong);
            tx.commit();
            return e;
        } catch (Exception e) {
            tx.rollback();
            logger.error(e.getLocalizedMessage());
            return Optional.empty();
        } finally {
            em.close();
        }
    }

    @Override
    public Iterable<Entity> findAll() {
        EntityManager em = emf.createEntityManager();
        EntityTransaction tx = em.getTransaction();

        try {
            tx.begin();
            Iterable<Entity> e = new EntityRepository(em).findAll();
            tx.commit();
            return e;
        } catch (Exception e) {
            tx.rollback();
            logger.error(e.getLocalizedMessage());
            return Collections.emptyList();
        } finally {
            em.close();
        }
    }

    @Override
    public Entity save(Entity entity) {
        EntityManager em = emf.createEntityManager();
        EntityTransaction tx = em.getTransaction();

        try {
            tx.begin();
            new EntityRepository(em).save(entity);
            tx.commit();
            return entity;
        } catch (Exception e) {
            tx.rollback();
            logger.error(e.getLocalizedMessage());
            return null;
        } finally {
            em.close();
        }
    }

    @Override
    public Entity update(Entity entity) {
        EntityManager em = emf.createEntityManager();
        EntityTransaction tx = em.getTransaction();

        try {
            tx.begin();
            new EntityRepository(em).update(entity);
            tx.commit();
            return entity;
        } catch (Exception e) {
            tx.rollback();
            logger.error(e.getLocalizedMessage());
            return null;
        } finally {
            em.close();
        }
    }

    @Override
    public boolean deleteOne(Long aLong) {
        EntityManager em = emf.createEntityManager();
        EntityTransaction tx = em.getTransaction();

        try {
            tx.begin();
            boolean r = new EntityRepository(em).deleteOne(aLong);
            tx.commit();
            return r;
        } catch (Exception e) {
            tx.rollback();
            logger.error(e.getLocalizedMessage());
            return false;
        } finally {
            em.close();
        }
    }

    @Override
    public long deleteAll() {
        EntityManager em = emf.createEntityManager();
        EntityTransaction tx = em.getTransaction();

        try {
            tx.begin();
            long r = new EntityRepository(em).deleteAll();
            tx.commit();
            return r;
        } catch (Exception e) {
            tx.rollback();
            logger.error(e.getLocalizedMessage());
            return 0;
        } finally {
            em.close();
        }
    }
}
